workflow/artifacts/s3/s3.go

```diff
@@ -38,13 +41,23 @@ func (s3Driver *S3ArtifactDriver) newS3Client() (argos3.S3Client, error) {
 func (s3Driver *S3ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
 	err := wait.ExponentialBackoff(wait.Backoff{Duration: time.Second * 2, Factor: 2.0, Steps: 5, Jitter: 0.1},
 		func() (bool, error) {
+			var err error
+			var bucket []byte
+			if inputArtifact.S3.BucketSecret.Key != "" {
```

Comment:
*Question/Suggestion* As not one super savvy in Go I am always unclear if this is best practice or if there is a better way to determine if something does not exist (rather than it being an empty string).

Reply:
hmm...I don't know, seems to be standard in codebase, but if someone knows better way, I'd love to know too

Reply:
you can check `len(x) > 0`, which will return `false` if `x` is either `nil` or the empty string